app/src/main/java/org/mozilla/fenix/ext/DownloadItem.kt

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ext
+
+import org.mozilla.fenix.R
+import org.mozilla.fenix.library.downloads.DownloadItem
+
+// While this looks complex, it's actually pretty simple.
+@SuppressWarnings("ComplexMethod")
+fun DownloadItem.getIcon(): Int {
+    fun getIconCornerCases(fileName: String?): Int {
+        return when {
+            fileName?.endsWith("apk") == true -> R.drawable.ic_file_type_apk
```

Comment:
@kglazko these checks are not very strong and fail on common filenames. for example `blahzip` is also recognized as a zip file and below `application/tartsaretasty` is recognized as a `ic_file_type_zip` because  it `contains` the substring `tar`.

It is better to be more precise for example by splitting the content type on `/` and looking for exact matches instead of `contains`. Similar with filenames which you can split on the final `.` and then look at the extension.

Reply:
That's a good catch! We would use `File("filename.ext").extension` for validating.